packages/pds/src/app-view/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -26,19 +34,21 @@ export default function (server: Server, ctx: AppContext) {
       const { ref } = db.dynamic
 
       // first verify there is not a block between requester & subject
-      const blocks = await ctx.services.appView
-        .graph(ctx.db)
-        .getBlocks(requester, actor)
-      if (blocks.blocking) {
-        throw new InvalidRequestError(
-          `Requester has blocked actor: ${actor}`,
-          'BlockedActor',
-        )
-      } else if (blocks.blockedBy) {
-        throw new InvalidRequestError(
-          `Requester is blocked by actor: $${actor}`,
-          'BlockedByActor',
-        )
+      if (requester) {
```

Comment:
a part of me just wants to two separate functions for "user getAuthorFeed" & "admin getAuthFeed" - not necessarily two different routes, just two functions that can call based on auth 🤔 

but also think this is totally fine & not that confusing. just depends on how much of a pattern we see this being going forward

Reply:
basically just that functions with a bunch of logic inside if statements are kinda hard to grok

Reply:
Yupyup!  I took a quick pass at this to extract some logic to help with readability, and then combined the two conditional blocks for access-based auth rules into a single block.